webapp/src/components/backstage/channel_selector.tsx

```diff
@@ -24,17 +24,35 @@ export interface Props {
 }
 
 const getMyPublicAndPrivateChannels = (state: GlobalState) => getMyChannels(state).filter((channel) =>
-    channel.type !== General.DM_CHANNEL && channel.type !== General.GM_CHANNEL,
+    channel.type !== General.DM_CHANNEL && channel.type !== General.GM_CHANNEL && channel.delete_at === 0,
 );
 
-type GetChannelType = (channelID: string) => Channel
+const filterChannels = (channelIDs: string[], channels: Channel[]): Channel[] => {
+    if (!channelIDs || !channels) {
+        return [];
+    }
 
-const ChannelSelector = (props: Props & { className?: string }) => {
+    const channelsMap = new Map<string, Channel>();
+    channels.forEach((channel: Channel) => channelsMap.set(channel.id, channel));
+
+    const result: Channel[] = [];
+    channelIDs.forEach((id: string) => {
+        let filteredChannel: Channel;
+        const channel = channelsMap.get(id);
+        if (channel && channel.delete_at === 0) {
+            filteredChannel = channel;
+        } else {
+            filteredChannel = {display_name: 'Unknown Channel', id} as Channel;
```

Comment:
```suggestion
            filteredChannel = {display_name: formatMessage({defaultMessage: 'Unknown Channel'}), id} as Channel;
```

Reply:
Else 
```suggestion
            filteredChannel = {display_name: '', id} as Channel;
```

With ... down in ChannelSelector
```tsx
                {channel.display_name || formatMessage({defaultMessage: 'Unknown Channel'})}
```